alacritty_terminal/src/grid/mod.rs

```diff
@@ -629,6 +689,23 @@ impl<'a, T> Iterator for GridIterator<'a, T> {
     }
 }
 
+impl<'a, T> DoubleEndedIterator for GridIterator<'a, T> {
```

Comment:
Why do we need the `DoubleEndedIterator` here, shouldn't the `BidirectionalIterator` work already? If this is a problem with iterating over all cells vs iterating over only empty cells, I think it would be a big mistake to have a `DoubleEndedIterator` which iterates over all, but a `BidirectionalIterator` which iterates only over non-empty cells.

Reply:
Calling `prev` does indeed work. I forget why I felt the need to write this separate iterator. Since I'm checking for `!cell.empty()` inside the `while let Some(cell) = iter.prev()` I don't think it was a functional reason like that.